app/src/marketplace/containers/EditProductPage/EditControllerProvider.jsx

```diff
@@ -230,6 +230,12 @@ function useEditController(product: Product) {
         const { beneficiaryAddress } = productRef.current
         if ((!address || isEthereumAddress(address)) && (!beneficiaryAddress || !areAddressesEqual(beneficiaryAddress, address))) {
             updateBeneficiaryAddress(address, false)
+
+            // save the new address immediately to db
+            await putProduct(State.update(productRef.current, () => ({
+                ...productRef.current,
+                beneficiaryAddress: address,
+            })), productRef.current.id || '')
```

Comment:
Should it still put the product into the db if `productRef.current.id` is nullish?

Reply:
basically it would never happen, it's just there to appease flow

Reply:
Good ol' flow…